File: 1984. Minimum Difference Between Highest and Lowest of K Scores.cpp
class Solution {
public:
    int minimumDifference(std::vector<int>& nums, int k) {
        std::sort(nums.begin(), nums.end());
        int left = 0, right = k;
        int ans = INT_MAX;
        while (right <= (int)nums.size()) {
            ans = std::min(ans, nums[right - 1] - nums[left]);
            left++;
            right++;
        }
        return ans;
    }
};

File: 1984. Minimum Difference Between Highest and Lowest of K Scores.java
class Solution {
    public int minimumDifference(int[] nums, int k) {
        Arrays.sort(nums);
        int left = 0, right = k;
        int ans = Integer.MAX_VALUE;
        while(right <= nums.length){
            ans = Math.min(ans, nums[right - 1] - nums[left]);
            left++;
            right++;
        }

        return ans;
    }
}